### PR TITLE
Add Charmhub documentation

### PR DESCRIPTION
**Pull request by dariofaccin**
# Description

Please include a summary of the change. Please also include relevant motivation and context. List any dependencies that are required for this change.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by gruyaume — CHANGES_REQUESTED** [review]
(same comment as in the self-signed-certificates PR)

Good job, this is very thorough.

However I think there is too much text. This is a fairly simple charm at the end of the day and we probably don't want 5-10 pages of text for it.

All the documentation that we add has to be maintained over time and the more concise we are, the less effort we'll have to place down the line.

We should also leverage all the automation we can. For instance, config and integrations are automatically generated in Charmhub. We shouldn't need to write our own docs about these subjects in addition to those.

In general I believe all of our documentation for this charm could hold on 1 page.

**Comment by dariofaccin** [comment]
@gruyaume I moved part of the tutorial (deploy, configure, integrate) into a single page. If you want ever less pages, I can moved them right in index page.
I kept the setup and cleanup pages, but these can be removed if you think so.
I removed the relation under "explanation" section since they are already linked in the index.
I would keep the "how to" page since the actions to manually provide a certificate need a precise execution order.
References (actions, configs, integrations, libraries, resources) are linked to the page automatically generated.

**Comment by saltiyazan** [comment]
Integration test issue addressed in the [this PR.](https://github.com/canonical/tls-certificates-operator/pull/44)
Take a look, you might want to rebase or wait for that one to be merged :)

File: docs/how-to/h-configure-per-unit.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# How to configure per-unit TLS certificates
```
Comment (gruyaume) [inline docs/how-to/h-configure-per-unit.md:1]:
```suggestion
# How to provide TLS certificates to requesting charms
```

File: docs/how-to/h-configure-per-unit.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+TLS Certificates Operator allows to configure different certificates for different units of the same desired application.
```
Comment (gruyaume) [inline docs/how-to/h-configure-per-unit.md:2]:
```suggestion
TLS Certificates Operator allows the provisioning of certificates to requiring charms.
```

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+The TLS Certificates Operator charm is responsible for distributing certificates through relationships. Certificates are provided by the operator through Juju configs.
```
Comment (gruyaume) [inline docs/index.md:1]:
```suggestion
The TLS Certificates Operator charm is responsible for distributing x.509 certificates in contexts where certificates are obtained through a manual process.
```

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+## Usage
+To deploy the TLS Certificates Operator, all you need to do is run the following command, which will fetch the charm from [Charmhub](https://charmhub.io/tls-certificates-operator) and deploy it to your model:
```
Comment (gruyaume) [inline docs/index.md:4]:
```suggestion
Deploy the charm:
```

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+```shell
+juju deploy tls-certificates-operator
+```
```
Comment (gruyaume) [inline docs/index.md:7]:
```suggestion
```shell
juju deploy tls-certificates-operator
juju status --watch 1s
```
```

File: docs/index.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+Juju will now fetch TLS Certificates Operator and begin deploying it to the local MicroK8s. This process can take several minutes depending on how provisioned (RAM, CPU, etc) your machine is. You can track the progress by running:
+```shell
+juju status --watch 1s
+```
+The deployment of the charm is completed when its status is in `Active/Idle` state.
```
Comment (gruyaume) [inline docs/index.md:13]:
```suggestion
```

File: docs/index.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+At this point you can relate the TLS Certificates Operator to your charm through the `tls-certificates` interface:
```
Comment (gruyaume) [inline docs/index.md:15]: Relate to a requirer of TLS certificates:

File: docs/index.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+```shell
+juju relate tls-certificates-operator <your-charm>
+```
+Wait for `juju status --watch 1s` to show all applications/units as `active`.
```
Comment (gruyaume) [inline docs/index.md:19]:
```suggestion
```

File: docs/index.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+TLS Certificates Operator is a member of the Ubuntu family. It’s an open source project that warmly welcomes community projects, contributions, suggestions, fixes and constructive feedback.
+
+* [Code of conduct](https://ubuntu.com/community/ethos/code-of-conduct)
+* Contribute and report bugs to the [charm](https://github.com/canonical/tls-certificates-operator)
+
+# Contents
+| Level | Path                 | Navlink                                                                                                                                                |
+|-------|----------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 1     | how-to               | [How-to guides]()                                                                                                                                      |
+| 2     | h-configure-per-unit | [Configure certificates per unit](/t/tls-certificates-operator-how-to-configure-per-unit-certificates/11612)                                           |
```
Comment (gruyaume) [inline docs/index.md:39]: I don't especially like the `per unit` wording that we have all over the place. Users use this charm whenever they obtain their certificates in a manual process. The per unit aspect is mostly an implementation detail, not what people care about.

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: docs/how-to/h-configure-per-unit.md (hunk 1 above)
Comment (dariofaccin) [reply docs/how-to/h-configure-per-unit.md]: Done.

File: docs/how-to/h-configure-per-unit.md (hunk 2 above)
Comment (dariofaccin) [reply docs/how-to/h-configure-per-unit.md]: Done.

File: docs/index.md (hunk 3 above)
Comment (dariofaccin) [reply docs/index.md]: Done.

File: docs/index.md (hunk 4 above)
Comment (dariofaccin) [reply docs/index.md]: Done.

File: docs/index.md (hunk 5 above)
Comment (dariofaccin) [reply docs/index.md]: Done.

File: docs/index.md (hunk 7 above)
Comment (dariofaccin) [reply docs/index.md]: Done.

File: docs/index.md (hunk 8 above)
Comment (dariofaccin) [reply docs/index.md]: Done.

File: docs/index.md (hunk 6 above)
Comment (dariofaccin) [reply docs/index.md]: Done.

File: docs/index.md (hunk 9 above)
Comment (dariofaccin) [reply docs/index.md]: I removed the `per unit` stuff and replaced with "Provisiong of certificates"

**Review by gruyaume — APPROVED** [review]

**Pull request merged**